Scheduled embedded user-task pull throws NPE when active user tasks have no matching subscription

## Description

When using the scheduled user-task pull with the Camunda 7 embedded adapter, a `NullPointerException` is thrown in `EmbeddedPullUserTaskDelivery.refresh()` as soon as there are active user tasks for which no matching
subscription exists.

Observed with:

- `process-engine-adapter-camunda-platform-c7-embedded-spring-boot-starter:2025.11.1`
- stack trace internally references `process-engine-adapter-camunda-platform-c7-embedded-core:2025.11.1`

## Stack trace excerpt

```java
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.concurrent.Future.get()" because "taskExecutionFuture" is null
    at dev.bpmcrafters.processengineapi.adapter.c7.embedded.task.delivery.pull.EmbeddedPullUserTaskDelivery.refresh$lambda$15(EmbeddedPullUserTaskDelivery.kt:116)
``` 

## Technical analysis

Based on code analysis, refresh() appears to create a Future for each active Camunda user task. If no matching TaskSubscriptionHandle is found for a task, null is returned. Afterwards, future.get() is still called
unconditionally.

Conceptually, the current flow seems to look like this:

```java
taskQuery.list()
    .parallelStream()
    .map { task ->
        val subscription = subscriptions.firstOrNull { matches(it, task) }
        if (subscription != null) executorService.submit { ... } else null
    }
    .forEach { future -> future.get() } // NPE if future == null
``` 

Additionally, forSubscriptions(...) seems not to actually restrict the query based on the available subscriptions, but only returns active(). Because of that, active user tasks are also considered even if they do not match any
registered subscription.

## Steps to reproduce

1. Register a user-task subscription, for example with a restriction on processDefinitionKey = A.
2. Have an active user task from a different process B in the same engine context.
3. Configure the user-task delivery strategy as embedded_scheduled.
4. Let the scheduler run.
5. EmbeddedPullUserTaskDelivery.refresh() throws a NullPointerException.

## Expected behavior

Active user tasks without a matching subscription should be ignored or logged cleanly, but they must not crash the scheduler.

## Actual behavior

The scheduled pull fails with a NullPointerException as soon as at least one active user task has no matching subscription.

## Suspected root cause

There are probably two issues:

1. refresh() does not filter out null futures before calling get().
2. forSubscriptions(...) does not constrain the TaskQuery using the subscription restrictions.

--- a/engine-adapter/c7-embedded-core/src/main/kotlin/dev/bpmcrafters/processengineapi/adapter/c7/embedded/task/delivery/pull/EmbeddedPullUserTaskDelivery.kt
+++ b/engine-adapter/c7-embedded-core/src/main/kotlin/dev/bpmcrafters/processengineapi/adapter/c7/embedded/task/delivery/pull/EmbeddedPullUserTaskDelivery.kt
@@ -51,8 +51,8 @@ class EmbeddedPullUserTaskDelivery(
         .initializeFormKeys()
         .forSubscriptions(subscriptions)
         .list()
-        .parallelStream()
-        .map { task ->
+        .asSequence()
+        .mapNotNull { task ->
           subscriptions
             .firstOrNull { subscription -> subscription.matches(task) }
             ?.let { activeSubscription: TaskSubscriptionHandle ->

--- a/engine-adapter/c7-embedded-core/src/test/kotlin/dev/bpmcrafters/processengineapi/adapter/c7/embedded/task/delivery/pull/EmbeddedPullUserTaskDeliveryTest.kt
+++ b/engine-adapter/c7-embedded-core/src/test/kotlin/dev/bpmcrafters/processengineapi/adapter/c7/embedded/task/delivery/pull/EmbeddedPullUserTaskDeliveryTest.kt
@@ -131,6 +131,32 @@ internal class EmbeddedPullUserTaskDeliveryTest {
     }
   }
 
+  @Test
+  fun `does not fail when task does not match any subscription`() {
+    // given
+    val deliveredDirectly = ConcurrentHashMap<String, TaskInformation>()
+
+    subscriptionRepository.createTaskSubscription(
+      TaskSubscriptionHandle(
+        taskType = TaskType.USER,
+        restrictions = mapOf(dev.bpmcrafters.processengineapi.CommonRestrictions.PROCESS_DEFINITION_KEY to "process-A"),
+        taskDescriptionKey = null,
+        payloadDescription = null,
+        action = { taskInformation, _ -> synchronized(deliveredDirectly) { deliveredDirectly[taskInformation.taskId] = taskInformation } },
+        termination = { }
+      )
+    )
+
+    // We need at least one task in the system to ensure that it doesn't match the subscription
+    assertThat(tasks).isNotEmpty
+
+    // when / then (should NOT throw NPE)
+    embeddedPullUserTaskDelivery.refresh()
+
+    // then
+    assertThat(deliveredDirectly).isEmpty()
+  }
+
   private fun randomTask() = TaskFake
     .builder()
     .id(UUID.randomUUID().toString())

--- a/engine-adapter/c7-remote-core/src/main/kotlin/dev/bpmcrafters/processengineapi/adapter/c7/remote/task/delivery/pull/PullServiceTaskDelivery.kt
+++ b/engine-adapter/c7-remote-core/src/main/kotlin/dev/bpmcrafters/processengineapi/adapter/c7/remote/task/delivery/pull/PullServiceTaskDelivery.kt
@@ -102,15 +102,16 @@ class PullServiceTaskDelivery(
 
     val taskActionHandlerCallables = lockedTasks
       .asSequence()
-      .map { lockedTask -> lockedTask to subscriptions.firstOrNull { subscription -> matches(lockedTask, subscription) } }
-      .filter { (lockedTask, subscription) ->
-        val keep = subscription != null
-        if (!keep) {
+      .mapNotNull { lockedTask ->
+        val subscription = subscriptions.firstOrNull { subscription -> matches(lockedTask, subscription) }
+        if (subscription != null) {
+          lockedTask to subscription
+        } else {
           metrics.incrementDroppedTasksCounter(lockedTask.topicName!!, NO_MATCHING_SUBSCRIPTIONS)
+          null
         }
-        keep
       }
-      .map { (lockedTask, activeSubscription) -> createTaskActionHandlerCallable(lockedTask, activeSubscription!!) }
+      .map { (lockedTask, activeSubscription) -> createTaskActionHandlerCallable(lockedTask, activeSubscription) }
       .toList()
 
     taskActionHandlerCallables.forEach { executor.submit(it) }

--- a/engine-adapter/c7-remote-core/src/main/kotlin/dev/bpmcrafters/processengineapi/adapter/c7/remote/task/delivery/pull/PullUserTaskDelivery.kt
+++ b/engine-adapter/c7-remote-core/src/main/kotlin/dev/bpmcrafters/processengineapi/adapter/c7/remote/task/delivery/pull/PullUserTaskDelivery.kt
@@ -54,8 +54,8 @@ class PullUserTaskDelivery(
       val taskDtoList = requireNotNull(result.body) { "Could not fetch user tasks for subscriptions: $subscriptions, status code was ${result.statusCode}" }
 
       taskDtoList
-        .parallelStream()
-        .map { task ->
+        .asSequence()
+        .mapNotNull { task ->
           subscriptions
             .firstOrNull { subscription -> subscription.matches(task) }
             ?.let { activeSubscription ->

--- a/engine-adapter/c7-remote-core/src/test/kotlin/dev/bpmcrafters/processengineapi/adapter/c7/remote/task/delivery/pull/PullUserTaskDeliveryTest.kt
+++ b/engine-adapter/c7-remote-core/src/test/kotlin/dev/bpmcrafters/processengineapi/adapter/c7/remote/task/delivery/pull/PullUserTaskDeliveryTest.kt
@@ -139,6 +139,29 @@ internal class PullUserTaskDeliveryTest {
     }
   }
 
+  @Test
+  fun `does not fail when task does not match any subscription`() {
+    // given
+    val deliveredDirectly = ConcurrentHashMap<String, TaskInformation>()
+
+    subscriptionRepository.createTaskSubscription(
+      TaskSubscriptionHandle(
+        taskType = TaskType.USER,
+        restrictions = mapOf(dev.bpmcrafters.processengineapi.CommonRestrictions.PROCESS_DEFINITION_KEY to "process-A"),
+        taskDescriptionKey = null,
+        payloadDescription = null,
+        action = { taskInformation, _ -> synchronized(deliveredDirectly) { deliveredDirectly[taskInformation.taskId] = taskInformation } },
+        termination = { }
+      )
+    )
+
+    // when
+    embeddedPullUserTaskDelivery.refresh()
+
+    // then
+    assertThat(deliveredDirectly).isEmpty()
+  }
+
   private fun randomTask() = TaskWithAttachmentAndCommentDto()
     .id(UUID.randomUUID().toString())
     .assignee("kermit")

